### ISSUE TITLE
[Feature request] mark discord sets as uncontrolled by seat

### ISSUE THREAD

**Issue by Nfinished** [warlof/seat-connector#60]
As an example I've got my security level set to strict, and autoassign connected users with alliance and corp discord roles. Any roles not explicitly assigned via access control in SeAT will be removed by the bot.

Today, if a user wants to join a ping group such as @mining, @ratting, etc, it requires an associated freejoin squad connected to discord roles. On a public server, this also immediately prevents any non-seat members from self-assigning the role.

Ideally, an admin could mark a group of discord roles 'unmanaged' or something similar, allowing users to self-assign/unassign discord roles by some other means like the popular bot reaction method.

**Comment by warlof** [comment]
This seems really discord specific

**Comment by Nfinished** [comment]
Yeah, I think so. Teamspeak already has channel roles which while not 1:1 with server roles are unmanaged by SeAT.

**Comment by Nfinished** [comment]
Looks like this has been discussed before in #16

**Comment by Bonsailinse** [comment]
One simple usecase is if you have roles that are not directly related to EVE Online. If I want to give social roles like "Other games", "Discord Moderator", "Twitch Subscriber", "CCP", other bots with their own roles, or really any role that is not related to the corporation itself I cannot have them below the bot role because it will remove them with every update.

So +1 for either having a way to manage a list of unmoderated roles or a mechanic where the bot only removes roles which are also given out by the bot.

To be honest this is a pretty major annoyance.

**Comment by Bonsailinse** [comment]
A much simpler edit would be if the function would not abort if one role cannot be unset.
This way we could put the role of the bot below the the ones that should be unmanaged.
The way error handling is done right now will prevent adding any role if something goes wrong.

That would not be pretty, but at least usable until the feature of uncontrolled roles are added.

Thanks!

**Comment by Nfinished** [comment]
Interesting solution but it could cause a conflict if the bot is used for anything else on the server.

**Comment by Bonsailinse** [comment]
You only would need to change the error handling in the [relevant function](https://github.com/warlof/seat-discord-connector/blob/5.0.x/src/Driver/DiscordMember.php#L202), don't see many problems relating other tasks the bot handles.

**Comment by Nfinished** [comment]
It'd by design prevent the same bot from managing seat and non-seat roles.

**Comment by Bonsailinse** [comment]
Not true, I just added a line to each function which skips the error handling so it just ignores if one role cannot be handled by the bot due to permissions. Now the function skips that role but still continues to try all other roles, so it adds/removes roles which are intended to be managed by SeAT.

This is by no means clean or a good way to handle it, but it was necessary for my server because the corp section is just an addition to a bigger community with a lot of manual managed roles.

**Comment by squigglezworth** [comment]
+1 for some way of handling this, please

**Comment by geraldhollis** [comment]
+1 for this for me too. Otherwise, this makes it difficult to have a combined Social/Eve Comms server.
